ocp_resources/resource.py

```diff
@@ -76,6 +83,29 @@ def _get_api_version(dyn_client: DynamicClient, api_group: str, kind: str) -> st
     return res.group_version
 
 
+@contextlib.contextmanager
+def get_retry_cluster_exceptions_client(
+    client: DynamicClient,
+    backoff_factor: float = 0.5,
+    retry_count: int = 5,
+    status_codes=None,
+) -> Any:
+    original_retries = client.configuration.retries
+    if not status_codes:
+        status_codes = set()
+    client.configuration.retries = urllib3.util.Retry(
+        total=retry_count,
+        backoff_factor=backoff_factor,
+        allowed_methods=(urllib3.util.Retry.DEFAULT_ALLOWED_METHODS | {http.HTTPMethod.PATCH, http.HTTPMethod.POST}),
+        status_forcelist=(DEFAULT_HTTP_STATUS_CODE_TO_RETRY | status_codes),
```

Comment:
the fact that `DEFAULT_HTTP_STATUS_CODE_TO_RETRY` is added by default is hidden here, please add to dosctring

@myakove do we want this behavior? i.e to force apply the defaults?

Reply:
@myakove / @rnetser please let me know how to proceed here.

Reply:
i'd add an option to use defaults